datadog-sidecar/src/service/sidecar_server.rs

```diff
@@ -297,12 +297,12 @@ impl SidecarServer {
 
             let futures = clients
                 .values()
-                .filter_map(|client| client.client.stats().ok())
+                .filter_map(|client| client.client.lock_or_panic().worker.stats().ok())
```

Comment:
maybe rename the `|client|`, `|telem_cached_entry|` to avoid the weird `client.client`